### DISCUSSION TITLE
从nodemodules引入formily/grid包的问题

### DISCUSSION THREAD

**Discussion by wd2012** [alibaba/formily#2391] [Q&A 2.x] [answered]
问题描述：
参照formily/antd中的formgrid组件写法，自己重写formgrid，但是当直接 import { Grid, IGridOptions } from '@formily/grid';时，maxColumns的设置无效，但是如果把@formily/grid的ts代码或编译后的js代码放到本地文件夹中、或在下图中useEffectLayout中设置一个无用的state时，可触发computed属性重新计算

```typescript
  const [test, setTest] = useState(false)

  useLayoutEffect(() => {
    // TODO: 这里的setTest操作是为了解决直接直接引用@formily/grid包时computed属性不能在依赖变化时自动更新问题
    // setTest(!test)
    return grid.connect(ref.current);
  }, [grid]);
```

**Comment by janryWang** [comment, marked as answer]
应该是重复打包了吧，grid包依赖了formily/reactive

**Reply by wd2012** [reply]
感谢，忽略了依赖，问题已解决

**Reply by zFitness** [reply]
怎么忽略依赖呢
